test/integration/master/kms_plugin_mock.go

```diff
@@ -34,7 +32,7 @@ import (
 
 const (
 	kmsAPIVersion = "v1beta1"
-	sockFile      = "/tmp/kms-provider.sock"
+	sockFile      = "@kms-provider.sock"
```

Comment:
should this be `/@kms-provider.sock` ?

Reply:
This is actually correct, because sockFile goes into net.Listen which is fine the @ sign in the file name.
In K8S encryption config, the endpoint should be provided in the format of unix:///temp/socket.sock. This endpoint is parsed via url.Parse, which fails to extract the path when supplied unix://@kms-plugin.sock. Hence, need to provide an extra / and the need for extra logic in the parseEndpoint to remove this extra /.
Not sure if there is a more elegant way to deal with this.
Note: the decision to use unix:///socke.path format was somewhat arbitrary.